sources/sql/src/lib.rs

```diff
@@ -106,64 +118,377 @@ fn rewrite_table_scans(plan: &LogicalPlan) -> Result<LogicalPlan> {
 
             match federated_source.as_any().downcast_ref::<SQLTableSource>() {
                 Some(sql_table_source) => {
-                    new_table_scan.table_name = TableReference::from(sql_table_source.table_name());
+                    let remote_table_name = TableReference::from(sql_table_source.table_name());
+                    known_rewrites.insert(original_table_name, remote_table_name.clone());
+
+                    // Rewrite the schema of this node to have the remote table as the qualifier.
+                    let new_schema = (*new_table_scan.projected_schema)
+                        .clone()
+                        .replace_qualifier(remote_table_name.clone());
+                    new_table_scan.projected_schema = Arc::new(new_schema);
+                    new_table_scan.table_name = remote_table_name;
                 }
                 None => {
                     // Not a SQLTableSource (is this possible?)
                     return Ok(plan.clone());
                 }
             }
 
-            // Wrap the table scan in a SubqueryAlias back to the original table name, so references continue to work.
-            let subquery_alias = LogicalPlan::SubqueryAlias(SubqueryAlias::try_new(
-                Arc::new(LogicalPlan::TableScan(new_table_scan)),
-                original_table_name,
-            )?);
-
-            return Ok(subquery_alias);
+            return Ok(LogicalPlan::TableScan(new_table_scan));
```

Comment:
Previously I was wrapping the table scan in an alias back to the original table name - which only works if the original table name was a bare reference (i.e no schema).